### PR TITLE
Delete Overlay Flag

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

**Comment by wu-hui** [comment]
cc @thebrianchen

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java [hunk 1]
```diff
@@ -297,24 +280,9 @@ void recalculateAndSaveOverlays(Set<DocumentKey> documentKeys) {
...
-      hardAssert(
-          fromOverlay.equals(fromMutationQueue),
-          "Documents from overlay do not match mutation queue version.");
-       */
-      return fromOverlay;
-    } else {
-      return getDocumentsMatchingCollectionQueryFromMutationQueue(query, offset);
-    }
+    ImmutableSortedMap<DocumentKey, Document> fromOverlay =
+        getDocumentsMatchingCollectionQueryFromOverlayCache(query, offset);
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java:284]: Can we inline this?

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java [hunk 2]
```diff
@@ -297,24 +280,9 @@ void recalculateAndSaveOverlays(Set<DocumentKey> documentKeys) {
...
-      /*
-      ImmutableSortedMap<DocumentKey, Document> fromMutationQueue =
-          getDocumentsMatchingCollectionQueryFromMutationQueue(query, sinceReadTime);
-      hardAssert(
-          fromOverlay.equals(fromMutationQueue),
-          "Documents from overlay do not match mutation queue version.");
-       */
-      return fromOverlay;
-    } else {
-      return getDocumentsMatchingCollectionQueryFromMutationQueue(query, offset);
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java:316]: This method can be removed.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java (hunk 1 above)
Comment (wu-hui) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java]: Done.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java (hunk 2 above)
Comment (wu-hui) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java:316]: Done.

**Review by schmidt-sebastian — APPROVED** [review]

**Review by thebrianchen — APPROVED** [review]
thanks for cleaning this up!

**Pull request merged**